t/40server_prepare.t

```diff
@@ -75,7 +75,9 @@ is_deeply($dbh->selectall_arrayref('SELECT id, mydata FROM t3'), [[1, 2]]);
 $dbh->{mariadb_server_prepare_disable_fallback} = 1;
 my $error_handler_called = 0;
 $dbh->{HandleError} = sub { $error_handler_called = 1; die $_[0]; };
-eval { $dbh->prepare("USE " . $dbh->quote_identifier($test_db)) };
+# MariaDB server since version 10.6.2 can prepare all statements except PREPARE, EXECUTE, and DEALLOCATE / DROP PREPARE. Previous MariaDB and Mysql versions cannot prepare USE statement
+my $non_preparable_statement = ($dbh->{mariadb_serverversion} >= 100602) ? q(PREPARE stmt FROM "SELECT 1") : ("USE " . $dbh->quote_identifier($test_db));
+eval { $dbh->prepare($non_preparable_statement); };
 $dbh->{HandleError} = undef;
 ok($error_handler_called, 'USE is not supported with mariadb_server_prepare_disable_fallback=1');
```

Comment:
The name of the test is misleading for 10.6.2+.

Also, "statements" is repeated twice in the commit description.